operations/codify-mgmt/oss/main.tf

```diff
@@ -0,0 +1,77 @@
+#------------------------------------------------------------------------------
+# The best practice is to use remote state file and encrypt it since your
+# state files may contains sensitive data (secrets).
+#------------------------------------------------------------------------------
+# terraform {
+#       backend "s3" {
+#             bucket = "remote-terraform-state-dev"
+#             encrypt = true
+#             key = "terraform.tfstate"
+#             region = "us-east-1"
+#       }
+# }
+
+
+# Use Vault provider
+provider "vault" { }
+
+#---------------------
+# Create policies
+#---------------------
+
+# Create admin policy in the root namespace
+resource "vault_policy" "admin_policy" {
+  name   = "admins"
+  policy = file("policies/admin-policy.hcl")
+}
+
+# Create 'training' policy
+resource "vault_policy" "eaas-client" {
+  name   = "eaas-client"
+  policy = file("policies/eaas-client-policy.hcl")
+}
+
+#--------------------------------
+# Enable userpass auth method
+#--------------------------------
+
+resource "vault_auth_backend" "userpass" {
+  type = "userpass"
+}
+
+# Create a user, 'student'
+resource "vault_generic_endpoint" "student" {
+  depends_on           = [vault_auth_backend.userpass]
+  path                 = "auth/userpass/users/student"
+  ignore_absent_fields = true
+
+  data_json = <<EOT
+{
+  "policies": ["eaas-client"],
+  "password": "changeme"
+}
+EOT
+}
+
+#----------------------------------------------------------
+# Enable secrets engines
+#----------------------------------------------------------
+
+# Enable K/V v2 secrets engine at 'kv-v2'
+resource "vault_mount" "kv-v2" {
+  path = "kv-v2"
+  type = "kv-v2"
+}
+
+# Enable Transit secrets engine at 'transit'
+resource "vault_mount" "transit" {
+  path = "transit"
+  type = "transit"
+}
+
+# Creating an encryption key named 'payment'
+resource "vault_transit_secret_backend_key" "key" {
+  depends_on = [vault_mount.transit]
+  backend    = "transit"
+  name       = "payment"
```

Comment:
```suggestion
  name       = "payment"  
  deletion_allowed = true
```

Reply:
This needs `deletion_allowed = true` or Terraform can't destroy this key to completely clean up